Description: Fix cblas_dspmv test failure on arm64
Origin: upstream, https://github.com/OpenMathLib/OpenBLAS/commit/0f9f6e4be55181237dd78adadb01b8aed4e7ec0f
Bug: https://github.com/OpenMathLib/OpenBLAS/issues/5708
Last-Update: 2026-04-03
---
This patch header follows DEP-3: http://dep.debian.net/deps/dep3/
diff --git a/kernel/arm64/dot_kernel_asimd.c b/kernel/arm64/dot_kernel_asimd.c
index 27f557c70..a1ae01f2d 100644
--- a/kernel/arm64/dot_kernel_asimd.c
+++ b/kernel/arm64/dot_kernel_asimd.c
@@ -262,7 +262,10 @@ USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 
 static RETURN_TYPE dot_kernel_asimd(BLASLONG n, FLOAT *x, BLASLONG inc_x, FLOAT *y, BLASLONG inc_y)
 {
-	volatile RETURN_TYPE  dot = 0.0;
+#ifndef DOUBLE
+    volatile
+#endif
+    RETURN_TYPE  dot = 0.0;
 	BLASLONG j = 0;
 
 	__asm__ __volatile__ (
